packages/medusa/src/api/admin/products/helpers.ts

```diff
@@ -70,26 +76,23 @@ export const remapVariantResponse = (variant: ProductVariantDTO) => {
       created_at: price.created_at,
       updated_at: price.updated_at,
     })),
-    price_set: undefined,
   }
+
+  delete (resp as any).price_set
+  return resp as any
 }
 
 export const refetchProduct = async (
   productId: string,
   scope: MedusaContainer,
   fields: string[]
 ) => {
-  const remoteQuery = scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
-  const queryObject = remoteQueryObjectFromString({
-    entryPoint: "product",
-    variables: {
-      filters: { id: productId },
-    },
-    fields: remapKeysForProduct(fields ?? []),
-  })
-
-  const products = await remoteQuery(queryObject)
-  return products[0]
+  return refetchEntity(
```

Comment:
It's not really needed since the caller to refetchProduct will call await, but I can actually remove the entire function since. I use refetchEntity now